frontend/public/components/marketplace/marketplace-page.tsx

```diff
@@ -0,0 +1,141 @@
+import {Helmet} from 'react-helmet';
+
+import {Firehose, PageHeading, StatusBox, MsgBox} from '../utils';
+import {referenceForModel, K8sResourceKind} from '../../module/k8s';
+import {PackageManifestModel, OperatorGroupModel, CatalogSourceConfigModel, SubscriptionModel} from '../../models';
+import {MarketplaceTileView} from './marketplace-items';
+import {PackageManifestKind, OperatorGroupKind, SubscriptionKind} from '../operator-lifecycle-manager';
+import {MARKETPLACE_CSC_NAME} from './index';
+import * as operatorImg from '../../imgs/operator.svg';
+
+const normalizePackageManifests = (packageManifests: PackageManifestKind[], kind: string) => {
+  const activePackageManifests = _.filter(packageManifests, packageManifest => {
+    return !packageManifest.status.removedFromBrokerCatalog;
+  });
+  return _.map(activePackageManifests, packageManifest => {
+    const name = packageManifest.metadata.name;
+    const uid = `${name}/${packageManifest.status.catalogSourceNamespace}`;
+    const defaultIconClass = 'fa fa-clone';
+    const iconObj = _.get(packageManifest, 'status.channels[0].currentCSVDesc.icon[0]');
+    const imgUrl = iconObj ? `data:${iconObj.mediatype};base64,${iconObj.base64data}` : operatorImg;
+    const iconClass = imgUrl ? null : defaultIconClass;
+    const provider = _.get(packageManifest, 'metadata.labels.provider');
+    const tags = packageManifest.metadata.tags;
+    const version = _.get(packageManifest, 'status.channels[0].currentCSVDesc.version');
+    const currentCSVAnnotations = _.get(packageManifest, 'status.channels[0].currentCSVDesc.annotations', {});
+    let {
+      description,
+      certifiedLevel,
+      healthIndex,
+      repository,
+      containerImage,
+      createdAt,
+      support,
+      longDescription,
+      categories,
+    } = currentCSVAnnotations;
+    const categoryArray = categories && _.map(categories.split(','), category => category.trim());
+    longDescription = longDescription ? longDescription : _.get(packageManifest, 'status.channels[0].currentCSVDesc.description');
+    const catalogSource = _.get(packageManifest, 'status.catalogSource');
+    const catalogSourceNamespace = _.get(packageManifest, 'status.catalogSourceNamespace');
+    return {
+      obj: packageManifest,
+      kind,
+      name,
+      uid,
+      iconClass,
+      imgUrl,
+      description,
+      provider,
+      tags,
+      version,
+      certifiedLevel,
+      healthIndex,
+      repository,
+      containerImage,
+      createdAt,
+      support,
+      longDescription,
+      categories: categoryArray,
+      catalogSource,
+      catalogSourceNamespace,
+    };
+  });
+};
+
+export const MarketplaceList: React.SFC<MarketplaceListProps> = (props) => {
+  const {packageManifest, loaded, loadError} = props;
+  const catalogSourceConfig = _.find(props.catalogSourceConfig.data, csc => csc.metadata.name === MARKETPLACE_CSC_NAME);
+  const items = loaded && packageManifest.loaded
```

Comment:
if packageManifest (i.e. if the pod crashed or something) is undefined this will give an error